cli/yamls.py

```diff
@@ -1,358 +1,217 @@
 
+    def __init__(self, message):
+        self.message = message
+        self.file_path = None
 
-def _limit_chars(_string, length):
-    length = int(length)
-    if length < 0:
-        raise exceptions.IRException('length to crop should be int, not ' +
-                                     str(length))
+    @classmethod
+    def from_yaml(cls, loader, node):
+        # Create & save references to Placeholder objects
+        placeholder = Placeholder(str(node.start_mark))
+        cls.placeholders_list.append(placeholder)
+        return placeholder
 
-    return _string[:length]
+    @classmethod
+    def to_yaml(cls, dumper, node):
+        message = re.sub("<string>", node.file_path, node.message)
+        raise exceptions.IRPlaceholderException(message)
 
 
-@configure.Configuration.add_constructor('limit_chars')
-def _limit_chars_constructor(loader, node):
+def load(settings_file, update_placeholders=True):
     """
-    Usage:
-        !limit_chars [<string>, <length>]
-    Method returns first param cropped to <length> chars.
-    """
-
-    params = loader.construct_sequence(node)
-    if len(params) != 2:
-        raise exceptions.IRException(
-            'limit_chars requires two params: string length')
-    return _limit_chars(params[0], params[1])
-
+    Loads and returns the content of a given YAML file as a dictionary
 
-@configure.Configuration.add_constructor('env')
-def _env_constructor(loader, node):
+    :param settings_file: String represents a path to a YAML file
+    :param update_placeholders: Whether to update Placeholder list or not in
+    case of Placeholder tag is in the given YAML file
+    :return: Dictionary containing the content the given YAML file
     """
-    usage:
-        !env <var-name>
-        !env [<var-name>, [default]]
-        !env [<var-name>, [default], [length]]
-    returns value for the environment var-name
-    default may be specified by passing a second parameter in a list
-    length is maximum length of output (croped to that length)
-    """
-
-    import os
-    # scalar node or string has no defaults,
-    # raise IRUndefinedEnvironmentVariableExcption if absent
-    if isinstance(node, yaml.nodes.ScalarNode):
-        try:
-            return os.environ[loader.construct_scalar(node)]
-        except KeyError:
-            raise exceptions.IRUndefinedEnvironmentVariableExcption(node.value)
+    LOG.debug("Loading setting file: %s" % settings_file)
+    if not os.path.exists(settings_file):
+        raise exceptions.IRFileNotFoundException(settings_file)
 
-    seq = loader.construct_sequence(node)
-    var = seq[0]
-    if len(seq) >= 2:
-        ret = os.getenv(var, seq[1])  # second item is default val
+    try:
+        with open(settings_file) as f_obj:
+            loaded_yml = yaml.load(f_obj)
 
-        # third item is max. length
-        if len(seq) == 3:
-            ret = _limit_chars(ret, seq[2])
-        return ret
+        # Handling case of empty file
```

Comment:
@aopincar if we do not allow empty yaml files, then we need to fix the installer/ospd/workarounds/rhos-8.0.yml file. This file is just a placeholder for future workarounds. 
Otherwise ospd installer for 8.0 will fail.